vscode/src/client.ts

```diff
@@ -379,6 +420,58 @@ export default class Client extends LanguageClient implements ClientInterface {
     }
   }
 
+  private async executeDelegateRequest(
+    type: string | MessageSignature,
+    params: any,
+  ): Promise<any> {
+    const request = typeof type === "string" ? type : type.method;
+    const originalUri = params.textDocument.uri;
+
+    // Delegating requests only makes sense for text document requests, where a URI is available
+    if (!originalUri) {
+      return null;
+    }
+
+    const hostLanguage = /\.([^.]+)\.erb$/.exec(originalUri)?.[1] || "html";
+    const vdocUriString = `embedded-content://${hostLanguage}/${encodeURIComponent(
+      originalUri,
+    )}.${hostLanguage}`;
+
+    if (request === "textDocument/completion") {
+      return vscode.commands
+        .executeCommand<CompletionList>(
```

Comment:
Call the completion provider for the delegated URI, which will invoke whatever language service is registered to handle it.